odoo/addons/base/models/ir_mail_server.py

```diff
@@ -511,6 +520,19 @@ def send_email(self, message, mail_server_id=None, smtp_server=None, smtp_port=N
             raise MailDeliveryException(_("Mail Delivery Failed"), msg)
         return message_id
 
+    @api.model
+    def _encapsulate_email(self, old_email, new_email):
+        """Change the FROM of the message and use the old one as name.
+        e.g.
+            Old From: "Admin" <admin@gmail.com>
+            New From: notifications@odoo.com
+            Output:   "Admin (admin@gmail.com)" <notifications@odoo.com>
+        """
+        if not old_email:
+            return new_email
+        old_email = old_email.replace('<', '(').replace('>', ')').replace('"', '')
```

Comment:
Why not using getaddresses ? Like email_split . In 12 we don't have all tools available, but we could at least do something like
```
old_email_split = getaddresses(old_email)
if not old_email_split or not old_email_split[0][1]:
    return old_email
email_as_name = "%s (%s)" % (old_email_split[0][0], old_email_split[0][1])
```

Reply:
the `replace()` is going to fail as soon as the name contains a reserved character (which is totally allowed), so we must use something smarter indeed ;)

Reply:
good idea :)